Readme.md

```diff
@@ -18,6 +18,23 @@ func main() {
 			fmt.Println("0 : err:", err)
 		})
 
+		GetUserName(5).Then(func(x interface{}) {
+			fmt.Println("5 : user:", x)
+			panic("a panic attack")
+		}).Catch(func(err error) {
+			fmt.Println("5 : err:", err)
+		})
+
+		// TODO: nested promises not very functional
+		GetUserName(7).Then(func(x interface{}) {
+			fmt.Println("7 (1): user:", x)
+		}).Then(func(x interface{}) {
+			fmt.Println("7 (2) : user:", x)
+			panic("another panic attack")
+		}).Catch(func(err error) {
+			fmt.Println("7 : err:", err)
+		})
```

Comment:
you can remove this from the readme and add a TODO section with this functionality listed 👍